sbomtool/internal/processor/processor.go

```diff
@@ -0,0 +1,119 @@
+// Package processor provides Syft-based SBOM processing capabilities optimized for Bottlerocket builds.
+// It configures Syft's catalogers for comprehensive package detection including Go and Rust binary analysis.
+package processor
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/format"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/sbom"
+)
+
+// GenerateComprehensiveSBOM generates an SBOM using configured Syft catalogers.
+// It includes comprehensive package detection with Go and Rust binary analysis.
+func GenerateComprehensiveSBOM(buildDir string) (*sbom.SBOM, error) {
+	// Create source from directory
```

Comment:
comment doesn't seem necessary - maybe "Create a syft source object that syft can scan for package discovery and SBOM generation"